nautobot/core/graphql/utils.py

```diff
@@ -1,6 +1,68 @@
+import graphene
+
+
+from django_filters.filters import BooleanFilter, NumberFilter, MultipleChoiceFilter
+from nautobot.utilities.filters import MultiValueNumberFilter
+
+
 def str_to_var_name(verbose_name):
     """Convert a string to a variable compatible name.
     Examples:
         IP Addresses > ip_addresses
     """
     return verbose_name.lower().replace(" ", "_").replace("-", "_")
+
+
+def get_filtering_args_from_filterset(filterset):
+    """Generate a list of filter arguments from a filterset.
+
+    The FilterSet class will be instantiated before extracting the list of arguments to
+    account for dynamic filters, inserted when the class is instanciated. (required for Custom Fields filters).
+
+    Filter field that are inheriting from BooleanFilter and NumberFilter will be converted
+    to their appropriate type, everything else will be of type String.
+    if the field field is a subclass of MultipleChoiceFilter, the argument will be converted as a list
+
+    Args:
+        filterset(FilterSet): FilterSet class used to extract the argument
+
+    Returns:
+        dict(graphene.Argument): Filter Arguments organized in a dictionary
+    """
+
+    args = {}
+    instance = filterset()
+
```

Comment:
I'm surprised that we're not calling graphene_django's `get_filtering_args_from_filterset` somewhere in this function to construct some baseline args - it seems like it's doing a lot with `filterset_class.base_filters` that we don't have in this function. Just want to make sure we're not losing functionality by rolling our own function here without calling the base implementation.

Reply:
As far as I've seen, the output generated by `get_filtering_args_from_filterset` wasn't that great because we are using a lot of nautobot specific filters, I've opened only one issue but I've found many more related to other field types like region because it's a MPTT model.
Another thing to consider is the output  of `get_filtering_args_from_filterset` is not easy to work with so if we call it first, we'll need to inspect what we are getting back to understand what is good and what needs to be cleaned up.